crypto/crypto/src/hkdf.rs

```diff
@@ -89,15 +91,16 @@ use thiserror::Error;
 #[derive(Clone, Debug)]
 pub struct Hkdf<D>
 where
-    D: Input + BlockInput + FixedOutput + Reset + Default + Clone,
+    D: Update + BlockInput + FixedOutput + Reset + Default + Clone,
+    D::BlockSize: ArrayLength<u8>,
```

Comment:
The more I look at this, the more I think this could leverage the [hkdf](https://github.com/RustCrypto/KDFs/tree/master/hkdf) crate more efficiently, and that not doing so leaves us with the burden of maintening this. Unless @kchalkias comes up with a good reaons why we're doing somethign fundamentally different here, could we open an issue to replace this with hkdf? Possibly label it help-wanted?

Reply:
There are two major updated to the original crate, this is one is generic for any hash function >=256 bits and does not allow empty messages + a few helper functions.

Reply:
@kchalkias - is the pr good to merge?   Would like to merge it before other dependency prs?